pkg/testcase/upgradenodereplacement.go

```diff
@@ -598,3 +598,25 @@ func buildRke2Cmd(
 
 	return cmd, nil
 }
+
+// DeleteReplacedNodes deletes the new nodes created as replacements.
+func DeleteReplacedNodes(cluster *shared.Cluster) {
+	externalIPs := shared.FetchNodeExternalIPs()
```

Comment:
you dont need this func since you have the cluster data 

```
   var ips []string
    ips = append(cluster.ServerIPs, ips...)
    ips = append(cluster.AgentIPs, ips...)


for _, ip : range ips ....

```

Reply:
this approach does not work as it tries to find the old ips, which got deleted as part of tf destroy...

Reply:
https://github.com/rancher/distros-test-framework/pull/166#discussion_r1788063328